base_vat_optional_vies/models/res_partner.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Tecnativa - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import re
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import vatnumber
+except ImportError:
+    _logger.warning(
+        "VAT validation partially unavailable because the `vatnumber` Python "
+        "library cannot be found. Install it to support more countries, "
+        "for example with `easy_install vatnumber` or "
+        "`pip install vatnumber`.")
+    vatnumber = None
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    vies_passed = fields.Boolean(
+        string="VIES validation passed", readonly=True)
+
+    def __init__(self, pool, cr):
+        super(ResPartner, self).__init__(pool, cr)
+        self._constraints = []
+
+    @api.constrains('vat')
+    def check_vat(self):
+        for partner in self:
+            if (not self.env.context.get('avoid_check_vat') and
```

Comment:
Call super always. In this case, you can call it when VIES bypass is not activated. Put also a mechanism for avoiding this special check when on test mode and not testing specifically this module. Use a similar technique than https://github.com/OCA/bank-payment/blob/c5b093fb5ef1df31a699e49c90db646db115b44a/account_payment_transfer_reconcile_batch/models/payment_order.py#L28